specification/synapse/data-plane/Microsoft.Synapse/preview/2020-08-01-preview/checkAccessSynapseRbac.json

```diff
@@ -0,0 +1,246 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "version": "2020-08-01-preview",
+    "title": "SynapseClient"
+  },
+  "x-ms-parameterized-host": {
+    "hostTemplate": "{endpoint}",
+    "useSchemePrefix": false,
+    "parameters": [
+      {
+        "$ref": "#/parameters/Endpoint"
+      }
+    ]
+  },
+  "schemes": [
+    "https"
+  ],
+  "paths": {
+    "/checkAccessSynapseRbac": {
+      "post": {
+        "tags": [
+          "CheckAccessSynapseRbac"
+        ],
+        "operationId": "CheckAccessSynapseRbac",
+        "description": "Check if the given principalId has access to perform list of actions at a given scope.",
+        "x-ms-examples": {
+          "Check access": {
+            "$ref": "./examples/CheckAccessSynapseRbac.json"
+          }
+        },
+        "consumes": [
+          "application/json",
+          "text/json"
+        ],
+        "produces": [
+          "application/json",
+          "text/json"
+        ],
+        "parameters": [
+          {
+            "name": "request",
+            "in": "body",
+            "required": true,
+            "description": "Details of scope, list of actions and principal.",

```

Comment:
I would flatten the parameter, I feel the experience would be better:

Example:
https://github.com/Azure/autorest.testserver/blob/edaa69641103edd0ba6f4e40384eefbc1becd574/swagger/model-flattening.json#L286